pkg/controller/pod.go

```diff
@@ -192,25 +205,105 @@ func (c *Controller) syncPod(ctx context.Context, pod *corev1.Pod) error {
 		return nil
 	}
 
+	if virtualNode.Data.Spec.ServiceDiscovery.AwsCloudMap.NamespaceName == nil ||
+		virtualNode.Data.Spec.ServiceDiscovery.AwsCloudMap.ServiceName == nil {
```

Comment:
i believe this should be return nil.  and log errors here via klog. (similar to other checks above).

I checked the source code of handler calling this handlePod, it will keep re-queue the pod in a backoff manner to retry it. so we should only return an error when it's a transilent error(e.g. can be retried).
In V1, we will handle this by validation webhooks, for now, return nil and log via klog should be enough

Reply:
Updated.